eclair-core/src/main/scala/fr/acinq/eclair/channel/fsm/Channel.scala

```diff
@@ -1127,12 +1365,35 @@ class Channel(val nodeParams: NodeParams, val wallet: OnChainChannelFunder with
       } else if (tx.txIn.map(_.outPoint.txid).contains(d.commitments.latest.fundingTxId)) {
         // counterparty may attempt to spend a revoked commit tx at any time
         handleRemoteSpentOther(tx, d)
+      } else if (d.commitments.all.exists(c => tx.txIn.map(_.outPoint.txid).contains(c.fundingTxId))) {
+        log.warning(s"a commit tx for an older commitment has been published txid=${tx.txid}")
+        // this was for another commitments
+        blockchain ! WatchAlternativeCommitTxConfirmed(self, tx.txid, nodeParams.channelConf.minDepthBlocks)
+        stay()
       } else {
         log.warning(s"unrecognized tx=${tx.txid}")
-        // this was for another commitments
         stay()
       }
 
+    case Event(WatchAlternativeCommitTxConfirmedTriggered(_, _, tx), d: DATA_CLOSING) =>
+      d.commitments.all
+        .find(c => tx.txIn.map(_.outPoint.txid).contains(c.fundingTxId)) match {
+        case Some(commitment) =>
+          log.info("hmm.... there has been a change of plan, a commit tx for fundingTxIndex={} fundingTxid={} has been confirmed", commitment.fundingTxIndex, commitment.fundingTxId)
+          val commitments1 = d.commitments.copy(
+            active = commitment +: Nil,
+            inactive = Nil
+          )
```

Comment:
The resulting `Commitments` instance is inconsistent, e.g. htlc indices, shachain won't match. It doesn't really matter but still.

Reply:
This is a very good point, it's worth diving into this. I believe the code here is incorrect (or at least, not optimal). The possible scenarios are:

- `commitment` was part of our `active` commitments: in that case everything matches and the current code works perfectly
- `commitment` was part of our `inactive` commitments:
  - if it has the same commitment number than the active commitments, no new changes have been made, and everything is consistent
  - if it does not have the same commitment number than the active commitments, we have received the revocation secret for that commitment, so we should treat it as a revoked commitment!

Reply:
Added a test on this specific case in 84ce26a79. It worked out of the box! :relaxed:

Reply:
Unfortunately, this test doesn't test the scenario that I'm suggesting, it's important that the revoked commitment was deactivated before making changes (and thus increasing the `fundingIndex`). I'm afraid it would *not* pass with the current code, but it's hard to test because we need to use zero-conf (which the current tests don't do).

Here is the exact layout of the test:

- initiate a first zero-conf splice
- save bob's corresponding commitment
- initiate a second zero-conf splice
- lock it on both sides, so that the first zero-conf splice becomes inactive
- make a payment: this will only be applied to the second zero-conf splice since the other one is inactive
- make bob publish his saved commitment from the first zero-conf splice (which is now actually revoked)

Reply:
Gotcha! Indeed there were issues. Added the test and fixes in f9b300a9b.